### ISSUE TITLE
Incorrectly configured webform generates a duplicated object 

### ISSUE THREAD

**Issue by mitchellkeaney** [esmero/webform_strawberryfield#41]
### What in tarnation?

I recently created a new webform from scratch. To my own fault, I did not follow [this documentation](https://github.com/esmero/archipelago-documentation/blob/8.x-1.0-beta2/docs/webformsasinput.md) which, had I done so, could have evaded hours spent scratching my head and running the same test over and over. 

However, in spite of negligence, something curious was uncovered. Before following the steps to [get the new webform to talk to SBF](https://github.com/esmero/archipelago-documentation/blob/8.x-1.0-beta2/docs/webformsasinput.md#setting-up-a-webform-to-talk-strawberryfield) - which was the entire section of this process that I had skipped - I would edit my existing content type (in this case, a Digital Object Collection) by adding some metadata via my new webform, save it, and then to my surprise a new Digital Collection Object would be created as a duplicate of the most recently edited (or created) Digital Object in my repository. This new Digital Object Collection would have all the same metadata and info as the Digital Object but when viewing this object, it looked the way a collection would be displayed (i.e. no images, no description, and a section for "Objects in this collection").

I'm not entirely sure exactly what this means, or why it was happening. @DiegoPino I know you probably have an answer or at least a better, logical understanding of why this would be. I'm guessing Archipelago, here, is trying to make sense of the new data given to it via a webform that is not configured to speak to SBF and therefore spits out a version of an existing object because that's what makes sense to it!

**Comment by DiegoPino** [comment]
@mitchellkeaney thanks! This is great. I mean, its weird because its a left over of a piece of functionality i coded but never exposed totally. This will lead into two tasks:
1.- Automatically generate settings needed, even if the users forgets to set them correctly (i also want to allow non AJAX form advancing, so some research needs to happen)
2.- Allowing this what you saw be used as a recover-from-backup functionality, since its really a automatic draft saving i have in place, which was permeating because the form was not doing its job. Good

**Comment by DiegoPino** [comment]
Adding to this. Also a skipped 'Edit metadata' click when using the not inline Strawberryfield widget will trigger this problem. Fix needs to come now as priority for Beta3.

**Comment by DiegoPino** [comment]
@mitchellkeaney @giancarlobi i will need some help with this one. I made a new Branch https://github.com/esmero/webform_strawberryfield/tree/ISSUE-41 that tries to solve this issue of an old webform session getting stuck, just to find more things i don't know what the user is expecting or not to happen and what we should or not provide. Could we have some talk about this tomorrow? Thanks!

**Comment by mitchellkeaney** [comment]
hey @DiegoPino sure thing. this afternoon works for me!

**Comment by DiegoPino** [comment]
@mitchellkeaney good. Related to #26  too of course. I think this really needs to be documented once for all and we need to set some clear expectations of what we can provide and not. I feel i made different decision based on use cases coming from not well justified/explained places and not its like a monster. Now language and UI are lacking. Hope we can get this back into shape. thx

**Comment by giancarlobi** [comment]
@DiegoPino Sorry, not sure to be available. I could comment your chat if report it here. In addition we have a very poor experience about webform so not sure I could be useful about this. Stay save.

**Comment by DiegoPino** [comment]
@giancarlobi thanks, no worries, i know its too late in the night our afternoon. What i want is to have a call with you/(Anna if she can, could be also a new call after ours) hopefully next week or when you both have time to find a workflow that makes sense regarding the Webform/ ADO lifecycle. Maybe you and i can meet next week? You guys all ok? Stay safe and let me know. Chat is also good here. Thanks amico 🇮🇹

**Comment by giancarlobi** [comment]
@DiegoPino Here all ok, thkx. I'll ask Anna for a meeting, good! Yes, I can next week, it could be starting from Monday, at the moment not so busy. Cuidate amigo!

**Comment by DiegoPino** [comment]
Things to check now:
- Why clear form is not working (when an unsaved draft/session is automatically loaded)
- Create a clear session link if a previous loaded sessions on top, but keep automatically loaded
- Validate the steps if there is data and create a new step for display settings?
- Change the word 'save metadata' to done with metadata, etc. Just not 'Save' so we don't have 2 saves and confuse the hell out of people

**Comment by mitchellkeaney** [comment]
Webform Workflow notes and suggestions:

- **The double _Save_ button**: Distinguishing "I'm done with entering Metadata" vs. "I'm ready to create my object". One potential option (though it has its own set of problems) is to move the "Save as" option to the last step in the Webform.
- **Where to put the object formatters**. These currently follow the user throughout the workflow ingest but they do not directly serve the metadata. Potential solutions could be 1) Moving these fields to be only in the last step in the ingest workflow. 2) Despite these fields not being metadata, we could create a step in the Webform that is just for editing the Revisions, Menu settings, Display settings, etc. (I thin this option then also relates to your 3rd bullet above @DiegoPino )
- **The "Descriptive Metadata" webform could (still) be more simple and clear** (this item may be better suited for issue #26 but will add it here for the time being)

1. To advance from steps _My Metadata_ and _Linked Data_ the user clicks a button titled "_Move on to next step_" but after that, the button switches to "_Next: Upload Files_" and the next step is actually titled "_Attach Files_". 
2. Depending on the media type, the _Attach Files_ step makes visible specific fields for uploading files. Example: selecting "Movie" will allow users to upload video files (m4v and mp4 by default) and images (jpg, jp2, tiff, etc. etc.) Yet, the name of these fields can be misleading: _Upload associated images_, _Upload associated documents_ and, actually, for media type "Movie" the upload field is just called _Videos_.

- **Saving previous webform session**: 1) Giving the users the option to "Clear" the Webform after it autofills, or 2) give them the option to view the old webform first before committing to picking up where the system remembers they left off. @DiegoPino we had talked internally about going with option 1, but wanted to document it here for future discussions. Something I'm thinking about now is, how will this logic work on a user to user basis...? For example: Mitchell is logged with his own user and starts working on an ingest and his computer crashes. Then Diego logs in with his own user and starts working on an ingest - does the system remember Mitchell's incomplete session and present it to Diego?

**Comment by DiegoPino** [comment]
Just as context, this https://github.com/esmero/webform_strawberryfield/issues/41#issuecomment-621453948 comes from a call we had with @mitchellkeaney yesterday regarding user assumptions and confusions about how webforms work regarding a full Node edit/creation when working with @daviserinanderson on a pilot for the studio.

@mitchellkeaney i see a lot of this proposals are things in your reach. Could you propose/share a fixed webform that tackles some of these so we can discuss/test the assumptions? A double sharing (here) but also implementing in studio.archipelago.nyc could bring more voices. What do you think?

**Comment by mitchellkeaney** [comment]
Definitely, and I agree! I will make some edits on my local Archipelago Descriptive Metadata and export the Webform to be tried and tested within studio.archipelago.nyc + sharing here.

**Comment by DiegoPino** [comment]
Pull is in #46
